coderd/database/dbmem/dbmem.go

```diff
@@ -2357,10 +2357,13 @@ func (q *FakeQuerier) DeleteOrganizationMember(ctx context.Context, arg database
 	q.mutex.Lock()
 	defer q.mutex.Unlock()
 
-	deleted := slices.DeleteFunc(q.data.organizationMembers, func(member database.OrganizationMember) bool {
-		return member.OrganizationID == arg.OrganizationID && member.UserID == arg.UserID
+	deleted := false
+	q.data.organizationMembers = slices.DeleteFunc(q.data.organizationMembers, func(member database.OrganizationMember) bool {
+		match := member.OrganizationID == arg.OrganizationID && member.UserID == arg.UserID
+		deleted = deleted || match
+		return match
 	})
-	if len(deleted) == 0 {
+	if !deleted {
```

Comment:
A dbmem bug I found

Reply:
oh this is nasty, took me a minute to realize what the bug was. 💀

if you want, rather than track `deleted` yourself, you could do something like

```go
	newMembers := slices.DeleteFunc(q.data.organizationMembers, func(member database.OrganizationMember) bool {
		return member.OrganizationID == arg.OrganizationID && member.UserID == arg.UserID
	})
	if len(newMembers) == len(q.data.organizationMembers) {
		return sql.ErrNoRows
	}
	q.data.organizationMembers = newMembers
```

Reply:
I had thought about that as well. That felt a bit less direct and less intuitive to me though, so I went with the crude and simple lol